### PR TITLE
[Android] Fix shadows size in clipped views

### LINKED ISSUE #11578: [Android] Shadow with clipping not working properly
### Description

When using shadow for view with clipping, the shadow bounds are not updated properly. 

Below is the code snippet used,
```
public CustomView()
{
    this.HeightRequest = 300;
    this.WidthRequest = 300;
    this.BackgroundColor = Colors.Lavender;
    this.Shadow = new Shadow();
    this.Shadow.Brush = new SolidColorBrush(Colors.Red);
    this.Clip = new RoundRectangleGeometry(new CornerRadius(20), new Rect(0, 0, this.HeightRequest, this.WidthRequest));
}
```
Android
Screenshot_1669206199

Windows
MicrosoftTeams-image

MAUI version used - 6.0.547

### Steps to Reproduce

Run the attached sample on the Android device
[ClipIssue.zip](https://github.com/dotnet/maui/files/10075528/ClipIssue.zip)

### Link to public reproduction project repository

Sample attached

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix shadows size in clipped views on Android.

Screenshot_1669222539

To validate the changes use the .NET MAUI Gallery and navigate to the Shadows page. If the shadow from the clipped view have the correct size, changes are validated.

image

### Issues Fixed

Fixes #11578

### CODE REVIEW

**Review by rmarinho — CHANGES_REQUESTED** [review]
Test is failing 
image

**Comment by jsuarezruiz** [comment]
> Test is failing
> image

Leave a small change behind, fixed.

**Comment by Redth** [comment]
@jsuarezruiz I pushed one small change to capture the Context and cache its use in a couple places in this view.

**Comment by jsuarezruiz** [comment]
> @jsuarezruiz I pushed one small change to capture the Context and cache its use in a couple places in this view.

Better, thanks.

**Comment by jsuarezruiz** [comment]
> > Test is failing
> > image
> 
> Leave a small change behind, fixed.

@rmarinho Could you review it again?. Thanks.

**Review by rachelkang — APPROVED** [review]
LGTM! Ready to go once all tests are passing

File: src/Core/src/Platform/Android/WrapperView.cs [hunk 1]
```diff
@@ -268,7 +270,8 @@ void DrawShadow(Canvas canvas)
 					else
 					{
 						var bounds = new Graphics.RectF(0, 0, canvas.Width, canvas.Height);
-						var path = Clip.PathForBounds(bounds)?.AsAndroidPath();
+						var density = _context.GetDisplayDensity();
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Core/src/Platform/Android/WrapperView.cs:273]: @Redth Thoughts?

[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Core/src/Platform/Android/WrapperView.cs]: Rui, reviewed and applied changes. Could you review it again?

**Review by rmarinho — APPROVED** [review]

**Pull request merged**